pkg/apiserver/profile/model.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profile
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/dbstore"
+)
+
+// TrackTickInterval is the interval to check the tasks status.
+const TrackTickInterval = 2 * time.Second
+
+// TaskState is used to represent the task/task group state.
+type TaskState string
+
+// Built-in task state
+const (
+	Create TaskState = "create"
+	Error  TaskState = "error"
+	Cancel TaskState = "cancel"
+
+	// TaskGroup can only have these two states.
+	Running TaskState = "running"
+	Finish  TaskState = "finish"
+)
+
+// TaskModel is the model definition of task.
+type TaskModel struct {
+	ID          string    `json:"task_id" gorm:"type:char(36);primary_key"`
+	TaskGroupID string    `json:"task_group_id" gorm:"type:char(36)"`
+	State       TaskState `json:"state"`
+	Addr        string    `json:"address"`
+	FilePath    string    `json:"file_path" gorm:"type:text"`
+	Component   string    `json:"component"`
+	CreateTime  int64     `json:"create_time"`
+	FinishTime  int64     `json:"finish_time"`
+	Error       string    `json:"error" gorm:"type:text"`
+}
+
+// TaskGroupModel is the model definition of task group.
+type TaskGroupModel struct {
+	ID           string    `json:"task_group_id" gorm:"type:char(36);primary_key"`
+	RunningTasks int       `json:"running_tasks"`
```

Comment:
Are you sure want to store the number of running tasks..

Reply:
for better tracing the running tasks